### PR TITLE
Cache relevance inside projections.

### PR DESCRIPTION
**Pull request by ppedrot**
This should speed up unimath by a few percents.

Overlays:
- https://github.com/MetaCoq/metacoq/pull/698

### CODE REVIEW

**Comment by ppedrot** [comment]
@coqbot bench

File: kernel/names.ml [hunk 1]
```diff
@@ -841,22 +842,36 @@ struct
...
+    let compare_gen a b =
+      let c = Int.compare a.proj_arg b.proj_arg in
+      if c <> 0 then c
+      else
+        let c = Int.compare a.proj_npars b.proj_npars in
+        if c <> 0 then c
+        else
+          let c = Label.compare a.proj_name b.proj_name in
+          if c <> 0 then c
+          else bcomp a.proj_relevant b.proj_relevant
```
Comment (SkySkimmer) [inline kernel/names.ml:864]: Can't you ignore proj_npars proj_name and proj_relevant here?

Comment (ppedrot) [reply kernel/names.ml:864]: Why would you? This could be a source of unsoundness if not used correctly. I think that this is a dangerous game.

**Comment by ppedrot** [comment]
@coqbot bench

**Comment by Alizter** [comment]
Slowdowns look like noise to me. Good job with those speedups.

**Comment by Janno** [comment]
What happened to coq-category-theory? The log says:
```
INFO: failed to install coq-category-theory
```

**Comment by SkySkimmer** [comment]
The OLD job for category theory said 
~~~
# Error: Cannot find a physical path bound to logical path
# Equations with prefix Equations.
~~~

**Comment by SkySkimmer** [comment]
test suite is broken too

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**